src/diagnostic.c

```diff
@@ -1,12 +1,264 @@
+    [YP_ERR_PIPEPIPEEQ_MULTI_ASSIGN]            = "Unexpected `||=` in a multiple assignment",
+    [YP_ERR_REGEXP_TERM]                        = "Expected a closing delimiter for the regular expression",
+    [YP_ERR_RESCUE_EXPRESSION]                  = "Expected a rescued expression",
+    [YP_ERR_RESCUE_MODIFIER_VALUE]              = "Expected a value after the `rescue` modifier",
+    [YP_ERR_RESCUE_TERM]                        = "Expected a closing delimiter for the `rescue` clause",
+    [YP_ERR_RESCUE_VARIABLE]                    = "Expected an exception variable after `=>` in a rescue statement",
+    [YP_ERR_RETURN_INVALID]                     = "Invalid `return` in a class or module body",
+    [YP_ERR_STRING_CONCATENATION]               = "Expected a string for concatenation",
+    [YP_ERR_STRING_INTERPOLATED_TERM]           = "Expected a closing delimiter for the interpolated string",
+    [YP_ERR_STRING_LITERAL_TERM]                = "Expected a closing delimiter for the string literal",
+    [YP_ERR_SYMBOL_INVALID]                     = "Invalid symbol", // TODO expected symbol? yarp.c ~9719
+    [YP_ERR_SYMBOL_TERM_DYNAMIC]                = "Expected a closing delimiter for the dynamic symbol",
+    [YP_ERR_SYMBOL_TERM_INTERPOLATED]           = "Expected a closing delimiter for the interpolated symbol",
+    [YP_ERR_TERNARY_COLON]                      = "Expected a `:` after the true expression of a ternary operator",
+    [YP_ERR_TERNARY_EXPRESSION_FALSE]           = "Expected an expression after `:` in the ternary operator",
+    [YP_ERR_TERNARY_EXPRESSION_TRUE]            = "Expected an expression after `?` in the ternary operator",
+    [YP_ERR_UNDEF_ARGUMENT]                     = "Invalid argument being passed to `undef`; expected a bare word, constant, or symbol argument",
+    [YP_ERR_UNARY_RECEIVER_BANG]                = "Expected a receiver for unary `!`",
+    [YP_ERR_UNARY_RECEIVER_MINUS]               = "Expected a receiver for unary `-`",
+    [YP_ERR_UNARY_RECEIVER_PLUS]                = "Expected a receiver for unary `+`",
+    [YP_ERR_UNARY_RECEIVER_TILDE]               = "Expected a receiver for unary `~`",
+    [YP_ERR_UNTIL_TERM]                         = "Expected an `end` to close the `until` statement",
+    [YP_ERR_WHILE_TERM]                         = "Expected an `end` to close the `while` statement",
+    [YP_ERR_WRITE_TARGET_READONLY]              = "Immutable variable as a write target",
+    [YP_ERR_WRITE_TARGET_UNEXPECTED]            = "Unexpected write target",
+    [YP_ERR_XSTRING_TERM]                       = "Expected a closing delimiter for the `%x` or backtick string",
+    [YP_WARN_AMBIGUOUS_FIRST_ARGUMENT_MINUS]    = "Ambiguous first argument; put parentheses or a space even after `-` operator",
+    [YP_WARN_AMBIGUOUS_FIRST_ARGUMENT_PLUS]     = "Ambiguous first argument; put parentheses or a space even after `+` operator",
```

Comment:
I know these have been extracted from the original source, but I cannot quite understand what `or a space even` part in these messages is supposed to mean. Are we trying to say "put parentheses or a space after `x` operator"?

Reply:
These warnings are printed in cases like:

```
a +c
```

I agree they're not very clear; however this particular error message is nearly identical to the CRuby parser error message:

https://github.com/ruby/ruby/blob/094f336a274ce83bc9868c7f9be0962b54ec46a7/parse.y#L8581

In any case, suggestions on how to clarify the warning would probably be welcome.